### ISSUE TITLE
Move all synchronisation functionalities into an extensions module, so to be optional

### ISSUE THREAD

**Issue by Mec-iS** [HTTP-APIs/hydrus#583]
### I'm submitting a 
- [x] feature request.

### Current Behaviour:
Socket functionalities automatically run when hydrus is started so it is ready for connections from the agent. Socket is readyed and all the helpers implement a synchronisation call on the socket to communicate with an agent.

### Expected Behaviour:
* hydrus should be started as "stand-alone" by default, support for the sync mechanism should be explicitly set to be available.
* Optimally synchornisation should be packaged in an external library to be installed as `pip install hydrus-agent-sync`.
* To achieve this decoupling, we should have a config variable to be set `USE_AGENT_SYNC=false/true` if `hydrus-agent-sync` is installed
*  also all the calls to socketio in the handlers should be factorised out in `hydrus-agent-sync` and called only if the synchronisation is on.

**Comment by sameshl** [comment]
Great, that you created this issue @Mec-iS. This decoupling was really needed in `hydrus`.

**Comment by chrizandr** [comment]
I agree, we want hydrus to be able to work without sync as well. Agent works right now without sync, this could be added here in a similar way
